segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/handler/AppendProcessorTest.java

```diff
@@ -339,7 +345,80 @@ public void testEventNumbers() {
 
         verifyNoMoreInteractions(store);
     }
-    
+
+    /**
+     * Test to ensure newer appends are processed only after successfully sending the DataAppended acknowledgement
```

Comment:
This test is good and helpful, but we also need a similar test where the first append is blocked when sending the DataAppened to the connection rather than when writing to the store.

Reply:
The current test simulates the blocking of `Append` when sending of `DataAppended` to connection is delayed/blocked.